reactive/bundleservice_charm.py

```diff
@@ -26,6 +25,11 @@ def setup_nagios(nagios):
     set_state('bundleservice-charm.nrpe-check-added')
 
 
+@when('website.available')
+def configure_website(website):
+    website.configure(port=hookenv.config('listen-port'))
+
```

Comment:
You might consider having this set a state like `website.configured` and then adding a `when_not` to this function on that state, and then creating the logic to unset the configured state when a relation parts.

On the other hand, the cost in this instance of redoing the configuration every time the hook fires is low, so you certainly don't *need* to.